### ISSUE TITLE
Mac OS build of client appears to have no accessibility menu bar label for screen readers

### ISSUE THREAD

**Issue by robin24** [owncloud/client#4174]
Hi,
I'm a fully blind Mac OS user and rely on the VoiceOver screen reader (built into Mac OS and iOS) in order to interface with the computer. The OC client setup screen is very accessible on Mac OS, however it is currently impossible to access and click the OwnCloud item in the status menu bar. Perhaps this is because the OwnCloud menu item has no text label (only represented as an image) or because an accessibility label is missing.
The issue is present in all current versions of the client, including ownCloud-2.0.2.2818 and the latest ownCloud-2.1.0.2890beta1.
It would be wonderful if this problem could be addressed, many thanks!
Further information regarding Mac OS accessibility, including VoiceOver as well as other accessibility features is available here:
http://www.apple.com/accessibility/osx/

**Comment by guruz** [comment]
Can't seem to get this to work with Qt 5.5.1 :-( As far as I see Qt does not have the code for this.

I can try to change Qt to provide this, stay tuned..

**Comment by guruz** [comment]
Trying to find it out.
https://stackoverflow.com/questions/35729889/os-x-nsstatusitem-how-to-set-the-accessibility-title-for-voiceover
https://forums.developer.apple.com/thread/27246

**Comment by guruz** [comment]
I have to conclude for now that this is not as easy as expected and would require some Qt modifications (switching over in ```qcocoasystemtrayicon.mm``` from the Qtish ```QNSImageView``` to use support for the OS X 10.10+ ```statusitem.button```.)..
I'll remove this from the 2.2.0 milestone for now, sorry.

It's really annoying that Apple does not do voiceover for their own applications in the menu bar either, so you have the "Menu, x items" for a lot of the tray applications.(only some of them say the real application name)

**Comment by robin24** [comment]
Hi Markus,
thanks a lot for your extensive research into this issue, it’s very greatly appreciated!
Yes, I agree that the status bar in OS X is not very consistent with VoiceOver, which is certainly a bug that Apple needs to fix.
I’m just wondering though, would it be possible to implement some kind of workaround in the meantime? For example, could there be a system-wide keyboard shortcut to open the OwnCloud preferences? For me this is the most annoying problem with the client, that I cannot access preferences in order to update account credentials, remove some folders from being synched, etc…
If there could be any kind of solution to bring up preferences accessibly (keyboard shortcut, terminal command, etc) this would already help me a lot…
Many thanks for the great support!
Robin
